src/app/pages/OpenWalletPage/__tests__/index.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { render, screen, waitFor } from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+import { when } from 'jest-when'
+import { selectShowAccountsSelectionModal } from 'app/state/importaccounts/selectors'
+import { canAccessNavigatorUsb } from 'app/lib/ledger'
+import { SelectOpenMethod } from '..'
+
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+}))
+jest.mock('app/lib/ledger')
+
+const mockNavigate = jest.fn()
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockNavigate,
+}))
+
+const renderComponent = (webExtensionLedgerAccess?: () => void) =>
+  render(
+    <MemoryRouter>
+      <SelectOpenMethod webExtensionLedgerAccess={webExtensionLedgerAccess} />
+    </MemoryRouter>,
+  )
+
+describe('<SelectOpenMethod />', () => {
+  beforeEach(() => {
+    jest.mocked(canAccessNavigatorUsb).mockResolvedValue(false)
+  })
+
+  it('should render component', async () => {
+    const { container } = await waitFor(() => renderComponent())
+
+    expect(container).toMatchSnapshot()
+  })
+
+  it('should render component with an access button', async () => {
+    await waitFor(() => renderComponent(() => {}))
+
+    expect(screen.queryByText('openWallet.method.ledger')).not.toBeInTheDocument()
+    expect(screen.getByText('ledger.extension.grantAccess')).toBeInTheDocument()
+  })
+
+  it('should redirect user to ledger page', async () => {
+    when(useSelector as any)
+      .calledWith(selectShowAccountsSelectionModal)
+      .mockReturnValue(true)
+
+    await waitFor(() => renderComponent(() => {}))
+
+    expect(mockNavigate).toHaveBeenCalledWith('/open-wallet/ledger')
+  })
+
+  it('should render variant with web usb support', async () => {
+    jest.mocked(canAccessNavigatorUsb).mockResolvedValue(true)
+    const { rerender } = await waitFor(() => renderComponent())
+    rerender(
+      <MemoryRouter>
+        <SelectOpenMethod />
+      </MemoryRouter>,
+    )
+
+    expect(screen.queryByText('errors.usbTransportNotSupported')).not.toBeInTheDocument()
```

Comment:
Add a test that establishes `screen.queryByText('errors.usbTransportNotSupported')` can sometimes be in the document. Otherwise `not.toBeInTheDocument` never detects if we renamed it, and error actually was shown

Reply:
That is handled by snapshot already.

Reply:
Yep yep, I'd simply add `expect(screen.getByText('errors.usbTransportNotSupported')).toBeInTheDocument()` after snapshot